src/modelinstance.cpp

```diff
@@ -1193,19 +1199,28 @@ Status ModelInstance::infer(float* data, float* output) {
     return StatusCode::OK;
 }
 
-Status ModelInstance::infer(const tensorflow::serving::PredictRequest* requestProto,
-    tensorflow::serving::PredictResponse* responseProto,
+template <typename RequestType, typename ResponseType>
+Status ModelInstance::infer(const RequestType* requestProto,
+    ResponseType* responseProto,
     std::unique_ptr<ModelInstanceUnloadGuard>& modelUnloadGuardPtr) {
     OVMS_PROFILE_FUNCTION();
     Timer<TIMER_END> timer;
     using std::chrono::microseconds;
 
-    auto status = validate(requestProto);
+    auto specialResources = getSR(requestProto, responseProto);  // request, response passed only to deduce type
```

Comment:
Can't we deduce from requestProto? It should be enough to distinguish between C-api/KFS/TFS

Reply:
You have to write to response so the processor needs both the request & response